service.go

```diff
@@ -321,6 +321,16 @@ type Interface interface {
 	Stop(s Service) error
 }
 
+// Shutdowner represents a service interface for a program that differentiates between "stop" and
+// "shutdown". A shutdown is triggered when the whole box (not just the service) is stopped.
+type Shutdowner interface {
+	Interface
+	// Shutdown provides a place to clean up program execution when the system is being shutdown.
+	// It is essentially the same as Stop but for the case where machine is being shutdown/restarted
+	// instead of just normally stopping the service.
+	Shutdown(s Service) error
```

Comment:
This is a fine implementation.

Please also indicate that if shutdown is called, that either: stop is also called after Shutdown, or the comment mentions that Stop will not be call if shutdown is called.

Otherwise LGTM.